querydsl-core/src/test/java/com/querydsl/core/types/dsl/PathBuilderTest.java

```diff
@@ -112,4 +117,27 @@ public void various() {
         entity.getTime("time", Time.class);
     }
 
+    @Test
+    public void attempting_to_overwrite_an_existing_path_with_one_of_the_same_type_succeeds() {
+        try {
+            PathBuilder<User> entity = new PathBuilder<User>(User.class, "entity");
+            String pathName = "some_path";
+            entity.get(pathName);
+            entity.get(pathName, Object.class);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void attempting_to_overwrite_an_existing_path_with_one_of_a_different_type_throws_a_RuntimeException() {
+        expectedException.expect(RuntimeException.class);
+        expectedException.expectMessage(
+                "Property name [some_path] of type [class java.lang.Object] cannot be overwritten with type [class java.lang.Integer] due to thread safety");
+        PathBuilder<User> entity = new PathBuilder<User>(User.class, "entity");
+        String pathName = "some_path";
+        entity.get(pathName);
+        entity.get(pathName, Integer.class);
```

Comment:
Let me sleep on it.
I think it could work, but there might be cases where this might throw too much in comparison to before.